examples/src/main/java/io/grpc/examples/header/CustomHeaderServer.java

```diff
@@ -53,9 +53,11 @@
   private Server server;
 
   private void start() throws Exception {
-    server = NettyServerBuilder.forPort(port).addService(ServerInterceptors
-            .intercept(GreeterGrpc.bindService(new GreeterImpl()), new HeaderServerInterceptor()))
-            .build().start();
+    server = ServerProvider.provider().builderForPort(port)
```

Comment:
No, you should use `ServerBuilder.forPort(port)`. ServerProvider is `@Internal`, so shouldn't be used directly.